### ISSUE TITLE
Graphviz files not parsable

### ISSUE THREAD

**Issue by LeifAndersen** [stchang/graph#25]
When a graph contains a bunch of structs (rather than just symbols or strings), the graphviz files produced by the `graphviz` call looks something like this:

```
digraph G {
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<sink-node>;
	#<source-node>;
	#<source-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	#<filter-node>;
	subgraph U {
		edge [dir=none];
	}
	subgraph D {
		#<filter-node> -> #<filter-node> [label=2];
		#<filter-node> -> #<filter-node> [label=1];
		#<filter-node> -> #<filter-node> [label=1];
		#<filter-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<sink-node> [label=+inf.0];
		#<source-node> -> #<filter-node> [label=+inf.0];
		#<source-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<filter-node> [label=1];
		#<filter-node> -> #<filter-node> [label=+inf.0];
		#<filter-node> -> #<filter-node> [label=1];
		#<filter-node> -> #<filter-node> [label=2];
	}
}
```

This is particularly problematic because `#<filter-node>` is not a valid node identifier in graphviz. In particular, when I run it, I get this error:

```
Error: :2: syntax error near line 2 context: >>> # <<< ; 
```

What would be better is give each node a unique (graphviz complaint) identifier, and use the `name` field to name the node.

**Comment by LeifAndersen** [comment]
Also, the graphviz function, as is, won't work if any two nodes have the same name.

**Comment by LeifAndersen** [comment]
Closed by #26
